sled-diagnostics/src/lib.rs

```diff
@@ -61,6 +61,18 @@ pub async fn dladm_info()
         .await
 }
 
+pub async fn nvmeadm_info()
+-> Vec<Result<SledDiagnosticsCmdOutput, SledDiagnosticsCmdError>> {
+    [nvmeadm_list()]
```

Comment:
Did you make these arrays in anticipation of running multiple nvmeadm, zfs, and zpool commands over time?
Otherwise you could just return a single result like above:
```rust
// List all zones on a sled.
pub async fn zoneadm_info()
-> Result<SledDiagnosticsCmdOutput, SledDiagnosticsCmdError> {
    execute_command_with_timeout(zoneadm_list(), DEFAULT_TIMEOUT).await
}
```

Reply:
Good point, I'll switch all of these for now. If/when we add additional commands we can update the API accordingingly.